service/history/queue/transfer_queue_processor_test.go

```diff
@@ -39,40 +57,926 @@ import (
 	"github.com/uber/cadence/service/worker/archiver"
 )
 
-func TestTransferQueueProcessor_RequireStartStop(t *testing.T) {
+type (
+	transferQueueProcessorSuite struct {
```

Comment:
Please avoid introducing new suites. As far as I understand, we generally agree that suites are not very useful, and you can achieve the same with one function that will prepare the object and mocks.

goleak.VerifyNone could be switched to VerifyTestMain and put into TestMain of the package.